utils/common/src/main/kotlin/FileMatcher.kt

```diff
@@ -39,13 +40,42 @@ class FileMatcher(
 ) {
     constructor(vararg patterns: String, ignoreCase: Boolean = false) : this(patterns.asList(), ignoreCase)
 
-    private val matcher = AntPathMatcher().apply {
-        setCaseSensitive(!ignoreCase)
+    companion object {
+        private val caseInsensitiveMatcher = AntPathMatcher().apply { setCaseSensitive(false) }
+        private val caseSensitiveMatcher = AntPathMatcher().apply { setCaseSensitive(true) }
+
+        /**
+         * Return true if [path] is matched case-insensitively by [pattern], false otherwise. The [path] must use '/' as
+         * separators, if it contains any.
+         */
+        fun matchCaseInsensitive(pattern: String, path: String) = caseInsensitiveMatcher.match(pattern, path)
```

Comment:
Just an idea for a bit shorter function names and for potentially avoiding case distinctions in the caller:`match(pattern: String, path: String, ignoreCase: Boolean = false)`?

Reply:
I wanted to avoid an if, but that was not necessary.